### PR TITLE
Remove dependency of `src/boot` on `src/shared`

### PR DESCRIPTION
**Pull request by robertknight**
This PR removes a dependency between code in `src/boot` on code in `src/shared`. This is important as we remove support for older (read: pre-ES 2015) browsers because code in `src/boot` still needs to work (or at least parse, not all will be executed) in those older browsers whereas code in `src/shared` does not. The code in `src/boot` still needs to run in these older browsers so that the client can gracefully fail startup (eg. with a console warning or UI notice).

- Rename and move the `jsonConfigsFrom` utility function from `src/shared/settings` to `parseJsonConfig` in `src/boot/parse-json-config`. Note this introduces a precedent that it is OK for code in `src/{shared, sidebar, annotator}` to depend on utilities in `src/boot`
- Move the polyfills code from `src/shared/polyfills` to `src/boot/polyfills`

### CODE REVIEW

File: src/boot/parse-json-config.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ function assign(dest, src) {
  *
  * @param {Document|Element} document - The root element to search.
  */
-export function jsonConfigsFrom(document) {
+export function parseJsonConfig(document) {
```
Comment (lyzadanger) [inline src/boot/parse-json-config.js:28]: I like the name change here.

File: src/annotator/config/settings.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import * as sharedSettings from '../../shared/settings';
+import { parseJsonConfig } from '../../boot/parse-json-config';
 
 import configFuncSettingsFrom from './config-func-settings-from';
 import isBrowserExtension from './is-browser-extension';
 
 export default function settingsFrom(window_) {
-  const jsonConfigs = sharedSettings.jsonConfigsFrom(window_.document);
+  const jsonConfigs = parseJsonConfig(window_.document);
```
Comment (lyzadanger) [inline src/annotator/config/settings.js:7]: As we make tests more modern and consistent, I wonder if we can knock it off with these test seams and just mock the needful things in the tests...but maybe there's another reason for the existence of `window_`?

File: src/boot/test/parse-json-config-test.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import { parseJsonConfig } from '../parse-json-config';
```
Comment (lyzadanger) [inline src/boot/test/parse-json-config-test.js:1]: Note for self: these tests were lifted from the (now-deleted) `settings-test` module.

Question for @robertknight Any meaningful changes from that set of tests?

**Review by lyzadanger — APPROVED** [review]
Once I got my head around what I believe to be the two most salient conceptual points:

* The code in `boot` must all be transpiled to be parse-able (if not executable in toto) by legacy browsers — it is more conservative than code elsewhere, and as such, cannot depend on any code from "outside of itself" as that code may not be transpiled conservatively enough;
* `boot` is responsible for handling polyfills

--which took rather longer than I'd like to admit, this looks fine to me. It runs without exploding locally.

File: src/annotator/config/settings.js (hunk 2 above)
Comment (robertknight) [reply src/annotator/config/settings.js:7]: Hmm ... maybe chat on this in Slack. Test seams do have some advantages over Sinon stubs so we may want to use different approaches in different scenarios.

File: src/boot/test/parse-json-config-test.js (hunk 3 above)
Comment (robertknight) [reply src/boot/test/parse-json-config-test.js:1]: No - the tests are the same as before except for the name change.

**Pull request merged**